core/templates/pages/preferences-page/preferences-page.component.ts

```diff
@@ -331,11 +315,86 @@ export class PreferencesPageComponent {
       this.loaderService.hideLoadingScreen();
     });
 
-    this.subjectInterestsChangeAtLeastOnce = false;
     this.TAG_REGEX_STRING = '^[a-z ]+$';
     this.LANGUAGE_CHOICES = this.languageUtilService.getLanguageIdsAndTexts();
     this.SITE_LANGUAGE_CHOICES = AppConstants.SUPPORTED_SITE_LANGUAGES;
   }
+
+  savePreferences(): void {
+    if (!this.hasUnsavedChanges) {
```

Comment:
Maybe use  (reactive-form)`FormGroup` and `FormControl` for the preference page to avoid so many management codes.

Reply:
I have made the functionality using a User Preferences model as suggested by Sean few days ago. PTAL at the changes. and if you still suggest FormGroup is needed here, I will do that.
Thanks!

Reply:
@AkashPaloju Do you think having FormGroup will reduce the complexity of this component? If yes, we can do it using FormGroup!

Reply:
@DubeySandeep sorry, busy due to exams, will reply very soon.

Reply:
Yes, It will reduce the complexity of this page, but I need to check, is it possible to make everything(types of input value) strict like now. Working on it. Will update very soon.

Reply:
@DubeySandeep PTAL, Thanks for this idea.